app.go

```diff
@@ -77,9 +77,13 @@ var (
 // to all other constructors, and called lazily at startup
 func (s *App) Provide(constructors ...interface{}) {
 	for _, c := range constructors {
-		// Print the constructor signature, file and line number from where it has come
-		file, line := funcLocation(c)
-		log.Printf("Providing constructor %q from %v:%d", reflect.TypeOf(c).String(), file, line)
+		if reflect.TypeOf(c).Kind() == reflect.Func {
+			// Print the constructor signature, file and line number from where it has come
+			file, line := funcLocation(c)
+			log.Printf("Providing constructor %q from %v:%d", reflect.TypeOf(c).String(), file, line)
+		} else {
+			log.Printf("Providing object %q", c)
```

Comment:
any reflect/runtime witchcraft can be done to see which file/package it came from? I think objects support `PkgPath()` on their types